src/System.Composition.Hosting/src/System/Composition/Hosting/Core/ExportDescriptorPromise.cs

```diff
@@ -83,9 +83,9 @@ public ExportDescriptor GetDescriptor()
             _creating = true;
             try
             {
-                var reply = _descriptor.Value;
-                Assumes.IsTrue(reply != null, "Export descriptor fulfillment function returned null.");
-                return reply;
+                ExportDescriptor relay = _descriptor.Value;
+                Requires.NotNull(relay, "getDescriptor");
```

Comment:
I would use descriptor as the param name.

From a compat perspective, I would try to keep the same exception here (InternalErrorException).